client/src/main/java/com/arcbees/beeshop/client/application/ApplicationView.java

```diff
@@ -137,25 +142,33 @@ public void f(Element element) {
         $(cartButton).click(new Function() {
             @Override
             public void f() {
-                if (shoppingCartOpen) {
-                    $(sidePanelContainer).removeClass(res.style().rightPanel__open());
-                    $(cartIcon).attr("class", font.icons().iconCart());
-                    $(cartButton).removeClass(res.style().active());
-                    $(cartIcon).text("");
-                    shoppingCartOpen = false;
-                } else {
-                    $(sidePanelContainer).addClass(res.style().rightPanel__open());
-                    $(cartIcon).text("+");
-                    $(cartIcon).removeClass("class", font.icons().iconCart());
-                    $(cartButton).addClass(res.style().active());
-                    shoppingCartOpen = true;
-                }
+                toggleCartVisibility();
             }
         });
 
         setI18nAnchors();
     }
 
+    private void toggleCartVisibility() {
+        if (shoppingCartOpen) {
+            closeCart();
+        } else {
+            $(sidePanelContainer).addClass(res.style().rightPanel__open());
+            $(cartIcon).text("+");
+            $(cartIcon).removeClass("class", font.icons().iconCart());
+            $(cartButton).addClass(res.style().active());
+            shoppingCartOpen = true;
+        }
+    }
+
+    private void closeCart() {
+        $(sidePanelContainer).removeClass(res.style().rightPanel__open());
+        $(cartIcon).attr("class", font.icons().iconCart());
```

Comment:
I think [I didn't write this] this is intended to remove all classes and add a new one. I guess it's to avoid doing `.removeClass()` and then `.addClass(x)` ?